bigcode_eval/evaluator.py

```diff
@@ -106,16 +106,23 @@ def evaluate(self, task_name, intermediate_generations=None):
 
             results, fine_grain_results = task.process_results(generations, references)
 
+            # Simplify the fine grain results to save task_id, generation, and whether it passed.
+            simplified_fine_grain_res = {}
+            for task_id in fine_grain_results:
+                # The element at each task_id is a single element list. The element is a tuple (0, dict) where the dict contains
+                # 'task_id', 'passed', 'result', and 'completion_id' as its keys. Extracting this dictionary to save. 
+                simplified_fine_grain_res[task_id] = fine_grain_results[task_id][0][1]
+
             if self.args.save_results_path:
                 # Dump to a new json that has {'task_id': --, 'generation': --, 'result': --}
-                save_results_path = f"{os.path.splitext(self.args.save_results_path)[0]}_{task_name}.json"
+                save_results_path = f"{self.args.save_results_path}_{task_name}.json"
```

Comment:
Can we just save these as files under that directory rather than using the path as a prefix? It's easier to manipulate directories than prefix groups.

Reply:
For this, I followed the way that it was done in the original codebase here: https://github.com/meher-m/bigcode-evaluation-harness/blob/main/main.py#L389. Do we want to maintain that choice? Or we could potentially change it in both places.

Reply:
Ah ok, I think they chose that method probably to maintain backward compatibility with some older version where they just emitted a file. Since we're adding a new code path here I think writing to a directory is easier to work with, but let's not change the existing code.